### PR TITLE
EasyBlock for Apache Serf, handled by SCons

### PR DESCRIPTION
**Pull request by hajgato**

### CODE REVIEW

**Comment by wpoely86** [comment]
looks good to me but I would promoted it to a generic SCONS easyblock? @hajgato up for that?

**Comment by boegel** [comment]
This looks *very* generic indeed, let make this a generic `SCons` easyblock (`easybuild/easyblocks/generic/scons.py`), as @wpoely86 suggested?

File: easybuild/easyblocks/s/serf.py [hunk 1]
```diff
@@ -0,0 +1,69 @@
+##
+# Copyright 2009-2015 Ghent University
```
Comment (boegel) [inline easybuild/easyblocks/s/serf.py]: new easyblock, so use `2015-2015`?

File: easybuild/easyblocks/s/serf.py [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+        pass
+
+    def build_step(self):
+        """
+        Build Serf using 'scons APR=/path/to/apr APU=/path/to/apu OPENSSL=/openssl/base PREFIX=/path/to/prefix
+        """
+        cmd = "scons %s PREFIX=%s" % (self.cfg['buildopts'], self.installdir)
+        (out, _) = run_cmd(cmd, log_all=True, simple=False)
+
+        return out
```
Comment (boegel) [inline easybuild/easyblocks/s/serf.py]: typically there's little reason to return the output here, it's not used 'upstream'

so, replace by

```python
run_cmd(cmd, log_all=True)
```

File: easybuild/easyblocks/generic/scons.py [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+class SCons(EasyBlock):
+    """Support for building/installing with SCons."""
+
+    def configure_step(self):
+        """
+        No configure step for SCons
+        """
+        pass
+
+    def build_step(self):
```
Comment (wpoely86) [inline easybuild/easyblocks/generic/scons.py]: let do this like in the `configuremake` block, add an arg `verbose=False` and add `log_output=verbose` to the `run_cmd`.

File: easybuild/easyblocks/generic/scons.py [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+# You should have received a copy of the GNU General Public License
+# along with EasyBuild.  If not, see <http://www.gnu.org/licenses/>.
+##
+"""
+EasyBuild support for building and installing SCons, implemented as an easyblock
+
+@author: Balazs Hajgato (Free University Brussels (VUB))
+"""
+
+import os
```
Comment (wpoely86) [inline easybuild/easyblocks/generic/scons.py]: why is this needed?

Comment (boegel) [inline easybuild/easyblocks/generic/scons.py]: +1, avoid unused imports

Comment (hajgato) [inline easybuild/easyblocks/generic/scons.py]: It was the original version, later the block changed a lot, and I forget to delete.

**Comment by hajgato** [comment]
@boegel: done

**Comment by wpoely86** [comment]
lgtm, ready to merge

**Comment by boegel** [comment]
Jenkins: test this please?

**Comment by boegel** [comment]
Jenkins: test this please?

**Comment by boegel** [comment]
Jenkins: test this please

**Comment by boegel** [comment]
Good to go in, thanks @hajgato!

File: easybuild/easyblocks/generic/scons.py [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+        (out, _) = run_cmd(cmd, log_all=True, log_output=verbose)
+
+        return out
+
+    def test_step(self):
+        """
+        Test with SCons 
+        """
+        cmd = "scons check"
+        (out, _) = run_cmd(cmd, log_all=True)
```
Comment (boegel) [inline easybuild/easyblocks/generic/scons.py]: @hajgato: can you issue a follow-up PR to only run the tests in case `self.cfg['runtest']` is `True`?

```python
if self.cfg['runtest']:
    (out, _) = run_cmd("scons check", log_all=True)
```

Comment (hajgato) [inline easybuild/easyblocks/generic/scons.py]: See PR #700
